CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeNodesBySuffixNGTest.java

```diff
@@ -0,0 +1,316 @@
+
+        // add vertices
+        vxId1 = graph.addVertex();
+        vxId2 = graph.addVertex();
+        vxId3 = graph.addVertex();
+        vxId4 = graph.addVertex();
+        vxId5 = graph.addVertex();
+        vxId6 = graph.addVertex();
+
+        // set the identifier of each vertex to something unique
+        graph.setStringValue(vertexIdentifierAttribute, vxId1, "d1");
+        graph.setStringValue(vertexIdentifierAttribute, vxId2, "One Id1");
+        graph.setStringValue(vertexIdentifierAttribute, vxId3, "Id2");
+        graph.setStringValue(vertexIdentifierAttribute, vxId4, "One Id2");
+        graph.setStringValue(vertexIdentifierAttribute, vxId5, "Pine Id2");
+        graph.setStringValue(vertexIdentifierAttribute, vxId6, "Id3");
+
+        // select all
+        graph.setBooleanValue(vertexSelectedAttribute, vxId1, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId2, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId3, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId4, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId5, true);
+        graph.setBooleanValue(vertexSelectedAttribute, vxId6, true);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of updateParameters method, of class MergeNodesBySuffix.
+     */
+    @Test
+    public void testUpdateParameters() {
+        System.out.println("testUpdateParameters");
+
+        final MergeNodesBySuffix instance = new MergeNodesBySuffix();
+
+        final PluginParameters parameters = new PluginParameters();
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> mergeTypeParameter = BooleanParameterType.build(MERGE_TYPE_PARAMETER_ID);
+        mergeTypeParameter.setBooleanValue(false);
+        parameters.addParameter(mergeTypeParameter);
+
+        final PluginParameter<IntegerParameterType.IntegerParameterValue> thresholdParameter = IntegerParameterType.build(THRESHOLD_PARAMETER_ID);
+        thresholdParameter.setIntegerValue(0);
+        parameters.addParameter(thresholdParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> mergeParameter = BooleanParameterType.build(MERGER_PARAMETER_ID);
+        mergeParameter.setBooleanValue(false);
+        parameters.addParameter(mergeParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> leadParameter = BooleanParameterType.build(LEAD_PARAMETER_ID);
+        leadParameter.setBooleanValue(false);
+        parameters.addParameter(leadParameter);
+
+        final PluginParameter<BooleanParameterType.BooleanParameterValue> selectedParameter = BooleanParameterType.build(SELECTED_PARAMETER_ID);
+        selectedParameter.setBooleanValue(false);
+        parameters.addParameter(selectedParameter);
+
+        final Map<String, PluginParameter<?>> parametersMap = parameters.getParameters();
```

Comment:
@Auriga2 all of this could be replaced with:

`final Map<String, PluginParameter<?>> parametersMap = parameters.createParameters();`

Reply:
Do you mean instance.createParameters()? But this plugin doesn't have a  `createParameters` method.
Parameters in `final PluginParameters parameters = new PluginParameters(); ` also doesn't have  a `createParameters` method.

Reply:
OK @Auriga2 you are right, ignore more.